sdc/datatypes/hpat_pandas_series_functions.py

```diff
@@ -4114,7 +4114,25 @@ def hpat_pandas_series_max_impl(self, axis=None, skipna=None, level=None, numeri
             _skipna = skipna
 
         if _skipna:
-            return numpy.nanmax(self._data)
+            input_arr = self._data
+            length = len(input_arr)
+
+            # set first element
+            result = numpy.nan
+            for i in range(length):
+                value = input_arr[i]
+                if not numpy.isnan(value):
+                    result = value
+                    break
+            else:
+                return result
+
+            for i in prange(length):
+                value = input_arr[i]
+                if not numpy.isnan(value):
+                    result = max(result, input_arr[i])
+
+            return result
```

Comment:
Could it be better?
`mask = numpy.isnan(self._data)
data_no_nan = self._data[~mask]
return data_no_nan.max()`

Reply:
It should be benchmarked.

Reply:
My version:
![image](https://user-images.githubusercontent.com/1541421/73353791-35ed4280-42a5-11ea-98ff-bfef47aa80a0.png)
@1e-to version:
![image](https://user-images.githubusercontent.com/1541421/73353807-43a2c800-42a5-11ea-9626-186a4a6310ab.png)

Reply:
> Could it be better?
> `mask = numpy.isnan(self._data) data_no_nan = self._data[~mask] return data_no_nan.max()`
```python
            mask = numpy.isnan(self._data)
            data_no_nan = self._data[~mask]
            return data_no_nan.max()
```

Reply:
It is expectedly that single parfor is faster. But Elena's version shows poor scalability. This is most likely due to `isnan` is not parallelized.

Reply:
@AlexanderKalistratov I will check `isnan`. I like this small code :)
What is `_data[~mask]`. Is it parallelized?

Reply:
I think it is. But I'm not sure.

The small code looks better, but it uses intermediate array and it would be slower anyway and uses more memory..
I'm afraid this time we should sacrifice readability to achieve better performance